docs/TOOLKIT.md

```diff
@@ -2,88 +2,77 @@
-The points are as such.
-
+## Turtle
+Our main drawing primitive is a turtle. It's a collection of polylines with an associated pen that has a location and direction.
+
+## Creating a Turtle
+**`const t = createTurtle();`**
+Initializes a new turtle object for drawing.
+> By default, the turtle object starts at points [0,0] with an angle of 0 degrees (bottom left corner, facing right) and begins in the drawing state.
+#### Rendering the Turtle
+-   **`drawTurtles([turtle])`** - Renders the specified list of turtles onto the canvas.
+## Drawing 
+ - **`t.goTo([ x: number, y: number ])`**  - Moves the turtle to the
+   specified coordinates, drawing a line (if the pen is down).
+ -  **`t.forward(distance: number)`** - Moves the turtle forward by the
+   specified distance, drawing a line (if the pen is down).
+-  **`t.arc(angle: number, radius: number)`** - Draws an arc with the given
+   angle **in degrees** and radius. The direction of the arc depends on
+   the sign of the angle (positive goes counterclockwise, negative goes
+   clockwise). 
+ - **`t.jump([x, y]):`** - Moves the turtle to specified
+   coordinates without drawing (basically goTo but always without
+   drawing).
+
+> All functions under `Drawing` end in the drawing state. 
+
+## Orientation and Movement
+- **`t.up()`** - Lifts the blot's pen.
+- **`t.down()`** - Lowers the blot's pen. 
+- **`t.setAngle(theta: number)`** - Sets the turtle's current direction to the specified angle **in degrees**. 
+- **`t.right(theta: number)`** - Rotates the turtle's direction to the right (clockwise) by the specified angle **in degrees**.
+- **`t.left(theta: number)`** - Rotates the turtle's direction to the left (counterclockwise) by the specified angle **in degrees**.
+ 
+ ## Transformations
+
+-   **`t.translate(from: pt, to: pt)`** - Translates the turtle's path from one point to another.
+-   **`t.rotate(angle: number, origin: pt)`** - Rotates the turtle's path around the specified origin point by the given angle **in degrees**.
+-   **`t.scale(factor: number, origin: pt)`** - Scales the turtle's path by the specified factor, relative to the given origin.
+> type pt = [ x: number, y: number ]
+
+## Importing SVGS
+**`t.fromSVG(svgString: string)`** - Imports an SVG
+> You can drag in SVG's, and the interface will generate a turtle for it. Keep in mind that SVG's are often imported far too large, and will need to be scaled and translated.
+
+## Special Points and Dimensions
+These are all variables. You can get 9 points of interest from the turtle as get methods.
+- **`t.start`** - Access the start points of the turtle's path.
+- **`t.end`** - Access the end points of the turtle's path.
+- **`t.width`** - Retrieve the width of the turtle's drawing area.
+- **`t.height`** -  Retrieve the height of the turtle's drawing area.
+> l is left, r is right, c is center, b is bottom.
+```js 
 lt -- ct -- rt
 |     |     |
 lc -- cc -- rc
 |     |     |
 lb -- cb -- rb
```

Comment:
Can we add something about what generative art is and how to make it? I couldn't find much about it on the blot website so it would be super cool to include a small blurb in the toolkit! Something like:

Generative art is art that has parts determined by a computer/autonomous system. For example, instead of you choosing where to draw a line, the computer selects a point using a random number generator. The Coding Train youtube channel is a really useful resource to learn more about generative art!